python/tests/test_1_online.py

```diff
@@ -1237,6 +1237,63 @@ def ac_outgoing_message(self, message):
     assert m == msg_in
 
 
+def test_reaction_to_partially_fetched_msg(acfactory, lp, tmpdir):
+    """See https://github.com/deltachat/deltachat-core-rust/issues/3688 "Partially downloaded messages are received out of order".
+
+    If the Inbox contains X small messages followed by Y large messages followed by Z small messages, Delta Chat first downloads a batch of X+Z messages, and then a batch of Y messages.
+
+    This bug was discovered by @Simon-Laux while testing reactions PR #3644 and can be reproduced with online test as follows:
+    - Bob enables download limit and goes offline.
+    - Alice sends a large message to Bob and reacts to this message with a thumbs-up.
+    - Bob goes online
+    - Bob first processes a reaction message and throws it away because there is no corresponding message, then processes a partially downloaded message.
+    - As a result, Bob does not see a reaction
+    """
```

Comment:
I think you should put some line breaks here, the lines are a lot longer than the rest of the code and hard to read on GitHub

Reply:
So, i decided to break lines after 100 chars. It's because my colums of code are 105-chars wide and if i add line breaks, say, after 120 chars, then i get two of them -- one by my Emacs line wrap and the original one -- and it looks ugly. So i'd prefer either to not break long lines at all or to make them short enough

Reply:
Yes, that's also about the maximum length of the other lines in this file